aws/usageReports/ebs/monthlyReport.go

```diff
@@ -0,0 +1,263 @@
+	}))
+	svc := ec2.New(sess)
+	desc := ec2.DescribeSnapshotsInput{SnapshotIds: []*string{aws.String(getResourceId(snap))}}
+	snapshots, err := svc.DescribeSnapshots(&desc)
+	if err != nil {
+		snapshotChan <- getSnapshotInfoFromES(ctx, snap, account, userId)
+		return err
+	}
+	for _, snapshot := range snapshots.Snapshots {
+		logger.Debug("DATA SNAPSHOT =======", map[string]interface{}{
+			"region": region,
+			"snap": snap,
+			"data": snapshot,
+		})
+		snapshotChan <- Snapshot{
+			SnapshotBase: SnapshotBase{
+				Id:          aws.StringValue(snapshot.SnapshotId),
+				Description: aws.StringValue(snapshot.Description),
+				State:       aws.StringValue(snapshot.State),
+				Encrypted:   aws.BoolValue(snapshot.Encrypted),
+				StartTime:   aws.TimeValue(snapshot.StartTime),
+				Region:      region,
+			},
+			Tags:  getSnapshotTag(snapshot.Tags),
+			Cost: snap.Cost,
+			Volume: Volume{
+				Id:   aws.StringValue(snapshot.VolumeId),
+				Size: aws.Int64Value(snapshot.VolumeSize),
+			},
+		}
+	}
+	return nil
+}
+
+// getEbsMetrics gets credentials, accounts and region to fetch EBS snapshots stats
+func fetchMonthlySnapshotsStats(ctx context.Context, snapshots []utils.CostPerResource, aa taws.AwsAccount, startDate time.Time) ([]SnapshotReport, error) {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	creds, err := taws.GetTemporaryCredentials(aa, MonitorSnapshotStsSessionName)
+	if err != nil {
+		logger.Error("Error when getting temporary credentials", err.Error())
+		return nil, err
+	}
+	defaultSession := session.Must(session.NewSession(&aws.Config{
+		Credentials: creds,
+		Region:      aws.String(config.AwsRegion),
+	}))
+	account, err := utils.GetAccountId(ctx, defaultSession)
+	if err != nil {
+		logger.Error("Error when getting account id", err.Error())
+		return nil, err
+	}
+	regions, err := utils.FetchRegionsList(ctx, defaultSession)
+	if err != nil {
+		logger.Error("Error when fetching regions list", err.Error())
+		return nil, err
+	}
+	snapshotChans := make([]<-chan Snapshot, 0, len(regions))
+	for _, snapshot := range snapshots {
+		for _, region := range regions {
+			if strings.Contains(snapshot.Region, region) {
+				snapshotChan := make(chan Snapshot)
+				go fetchMonthlySnapshotsList(ctx, creds, snapshot, account, region, snapshotChan, aa.UserId)
+				snapshotChans = append(snapshotChans, snapshotChan)
+			}
+		}
+	}
+	snapshotsList := make([]SnapshotReport, 0)
+	for snapshot := range merge(snapshotChans...) {
+		snapshotsList = append(snapshotsList, SnapshotReport{
+			ReportBase: utils.ReportBase{
+				Account:    account,
+				ReportDate: startDate,
+				ReportType: "monthly",
+			},
+			Snapshot: snapshot,
+		})
+	}
+	return snapshotsList, nil
+}
+
+// filterSnapshotsCosts filters snapshots, cloudwatch and volumes of EBS snapshots costs
+func filterSnapshotsCosts(ebsCost, cloudwatchCost []utils.CostPerResource) ([]utils.CostPerResource, []utils.CostPerResource, []utils.CostPerResource) {
```

Comment:
This function is not used. Do we need to keep it ?

Reply:
Fixed